src/app/child-dev-project/children/children-list/children-list.component.ts

```diff
@@ -257,49 +236,25 @@ export class ChildrenListComponent implements OnInit, AfterViewInit {
     this.centerFS.options = options;
   } */
 
-  prepareAttendanceData(loadedEntities: AttendanceMonth[]) {
-    this.attendanceList = new Map<string, AttendanceMonth[]>();
-    loadedEntities.forEach((x) => {
-      if (!this.attendanceList.has(x.student)) {
-        this.attendanceList.set(x.student, new Array<AttendanceMonth>());
-      }
-      this.attendanceList.get(x.student).push(x);
-    });
-
-    this.attendanceList.forEach((studentsAttendance) => {
-      studentsAttendance.sort((a, b) => {
-        // descending by date
-        if (a.month > b.month) {
-          return -1;
-        }
-        if (a.month < b.month) {
-          return 1;
-        }
-        return 0;
-      });
-    });
-  }
-
   applyFilter(filterValue: string) {
     filterValue = filterValue.trim();
     filterValue = filterValue.toLowerCase(); // MatTableDataSource defaults to lowercase matches
     this.childrenDataSource.filter = filterValue;
   }
 
   displayColumnGroup(columnGroupName: string) {
-    this.columnGroupSelection = columnGroupName;
-    let found = false;
-    let group: ColumnGroup;
-    let i = 0;
-    while (!found && i < this.columnGroups.length) {
-      if (this.columnGroups[i].name === columnGroupName) {
-        found = true;
-        group = this.columnGroups[i];
-      }
-      i++;
+    // When components, that are used in the list (app-list-attendance), also listen to the mediaObserver, a new
+    // mediaChange is created once this used component is displayed (through column groups change). This may
+    // re-trigger the settings for small screens. Therefore, we only allow a change ever 0.5 seconds to prevent this.
+    if (this.ready) {
```

Comment:
we really need to find a cleaner solution to deal with the resizing ...

Reply:
Yes, I don't like this either. The resizing configurations are scattered across the whole application. I would like to define it only once per view (Dashboard, School, ...).